runtime/tchannel_client.go

```diff
@@ -277,7 +275,11 @@ func (c *tchannelOutboundCall) finish(err error) {
 	c.metrics.Latency.Record(c.finishTime.Sub(c.startTime))
 
 	// write logs
-	c.logger.Info("Finished an outgoing client TChannel request", c.logFields()...)
+	if err == nil {
+		c.logger.Info("Finished an outgoing client TChannel request", c.logFields()...)
+	} else {
+		c.logger.Warn("Failed to send outgoing client TChannel request", zap.Error(err))
```

Comment:
should this be logger.error

Reply:
This is open to discussion. My reasoning is that when error happens, we will look at one error message (which is logged at endpoint handler and anywhere else is warn) and could understand what happened, if that is not sufficient we then look further for the warn logs associated with the request. Based on what I have experienced, multiple error log sites that describe the same cause create distraction and make logs less effective.

Reply:
We could have a zap.field GUID associated with each warn, and let the error catch that GUID which is the root cause of this Error. We could have a runbook/wiki document all the mitigation step for each GUID and handle the oncall to SRE

Reply:
Once we use context.logger everywhere, we will have the request uuid in the context and all the logs that associated with the request will have the same request uuid.